File: weather-mvp-dagger/src/main/java/com/baronzhang/android/weather/main/WeatherApplication.java
package com.baronzhang.android.weather.main;

import android.app.Application;
import android.content.Context;
import android.os.StrictMode;
import android.util.Log;

import com.baronzhang.android.weather.BuildConfig;
import com.baronzhang.android.weather.di.ApplicationComponent;
import com.baronzhang.android.weather.di.ApplicationModule;
import com.baronzhang.android.weather.di.DaggerApplicationComponent;

/**
 * @author baronzhang (baron[dot]zhanglei[at]gmail[dot]com)
 * 16/2/4
 */
public class WeatherApplication extends Application {
    private static final String TAG = "WeatherApp";
    private ApplicationComponent applicationComponent;
    private static WeatherApplication application;

    public static WeatherApplication getInstance() {
        return application;
    }

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        Log.d(TAG, "attachBaseContext");
    }

    @Override
    public void onCreate() {
        super.onCreate();
        Log.d(TAG, "onCreate start");
        application = this;
        if (BuildConfig.DEBUG) {
            StrictMode.setThreadPolicy(new StrictMode.ThreadPolicy.Builder().detectAll().penaltyLog().build());
            StrictMode.setVmPolicy(new StrictMode.VmPolicy.Builder().detectAll().penaltyLog().build());
        }
        applicationComponent = DaggerApplicationComponent.builder()
                .applicationModule(new ApplicationModule(this))
                .build();

        //初始化Stetho
        BuildConfig.STETHO.init(getApplicationContext());
        Log.d(TAG, "onCreate end");
    }


    public ApplicationComponent getApplicationComponent() {
        return applicationComponent;
    }
}

File: weather-mvp-dagger-kotlin/src/main/java/com/minimalist/weather/kotlin/main/WeatherApplication.kt
package com.minimalist.weather.kotlin.main

import android.app.Application
import android.content.Context
import android.util.Log
import com.minimalist.weather.kotlin.BuildConfig
import com.minimalist.weather.kotlin.di.ApplicationComponent
import com.minimalist.weather.kotlin.di.ApplicationModule
import com.minimalist.weather.kotlin.di.DaggerApplicationComponent

class WeatherApplication : Application() {
    var applicationComponent: ApplicationComponent? = null
        private set

    override fun attachBaseContext(base: Context) {
        super.attachBaseContext(base)
        Log.d(TAG, "attachBaseContext")
    }

    override fun onCreate() {
        super.onCreate()
        Log.d(TAG, "onCreate start")
        instance = this
//        if (BuildConfig.DEBUG) {
//            StrictMode.setThreadPolicy(StrictMode.ThreadPolicy.Builder().detectAll().penaltyLog().build())
//            StrictMode.setVmPolicy(StrictMode.VmPolicy.Builder().detectAll().penaltyLog().build())
//        }
        applicationComponent = DaggerApplicationComponent.builder()
                .applicationModule(ApplicationModule(this))
                .build()

        //初始化Stetho
        BuildConfig.STETHO.init(applicationContext)
        Log.d(TAG, "onCreate end")
    }

    companion object {
        private val TAG = "WeatherApp"
        lateinit var instance: WeatherApplication
            private set
    }
}
